tensorflow/compiler/tf2tensorrt/convert/ops/selectv2.cc

```diff
@@ -89,32 +115,105 @@ class ConvertSelectV2 : public OpConverterBase<ConvertSelectV2> {
         *tensor_[0]->trt_tensor(),  // cond_tensor
         *tensor_[1]->trt_tensor(),  // then_tensor
         *tensor_[2]->trt_tensor()   // else_tensor
-    );
+        );
 
-    converter->SetLayerName(select_layer, params.node_def.name(), "selectv2");
+    converter->SetLayerName(select_layer, params.node_def.name(), layer_name_);
     AddOutput(TRT_TensorOrWeights(select_layer->getOutput(0)));
     return OkStatus();
   }
 
  private:
+  Status compare_shapes(const nvinfer1::Dims& shape1,
+                        const nvinfer1::Dims& shape2,
+                        bool* cond_is_vector = nullptr) const {
+    const bool then_vs_else = cond_is_vector == nullptr;
+    bool same_shapes = shape1 == shape2;
+    if (!same_shapes && shape1.nbDims == shape2.nbDims) {
+      // We can't check size equivalent when dynamic shapes are involved.
+      // In this case, the two shapes should be equal at runtime. Therefore,
+      // the shapes still should be considered as equal if at least one of
+      // them is a tensor with dynamic shape,
+      same_shapes = DynamicShapeInput(this->params_->inputs, then_vs_else);
+    }
+    if (!same_shapes) {
+      if (then_vs_else ||
+          !(*cond_is_vector =
+                (shape1.nbDims == 1 && shape1.d[0] == shape2.d[0]))) {
+        const auto err = input_shapes_error_msg(
+            shape1, shape2, this->params_->node_def, then_vs_else);
+        return errors::InvalidArgument(err);
+      }
+    }
+    return OkStatus();
+  }
+
+  bool DynamicShapeInput(const std::vector<TRT_TensorOrWeights>& inputs,
+                         bool then_vs_else) const {
+    const int idx = then_vs_else ? 1 : 0;
+    for (int i = 0; i < 2; ++i) {
+      const auto& input = inputs.at(i + idx);
+      if (input.is_tensor()) {
+        const auto* dims = input.GetTrtDims().d;
+        for (int j = input.GetTrtDims().nbDims; --j >= 0;) {
+          if (*(dims + j) < 0) return true;
+        }
+      }
```

Comment:
Please use the existing helper function:
```suggestion
      if (input.is_tensor() && !HasStaticShape(inputs.GetTrtDims())) {
        return true;
      }
```

Reply:
Done.